ui/app/controllers/optimize.js

```diff
@@ -1,27 +1,41 @@
 import Controller from '@ember/controller';
-import { tracked } from '@glimmer/tracking';
-import { sort } from '@ember/object/computed';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { inject as controller } from '@ember/controller';
 import { task, timeout } from 'ember-concurrency';
-import Ember from 'ember';
 
 export default class OptimizeController extends Controller {
-  @tracked recommendationSummaryIndex = 0;
-
-  summarySorting = ['submitTime:desc'];
-  @sort('model', 'summarySorting') sortedSummaries;
+  @service router;
+  @controller('optimize/summary') summaryController;
 
   get activeRecommendationSummary() {
-    return this.sortedSummaries.objectAt(this.recommendationSummaryIndex);
+    const currentRoute = this.router.currentRoute;
+
+    if (currentRoute.name === 'optimize.summary') {
+      return this.summaryController.model;
+    } else {
+      return undefined;
+    }
   }
 
   @(task(function*() {
-    this.recommendationSummaryIndex++;
+    const currentSummaryIndex = this.model.indexOf(this.activeRecommendationSummary);
+    const nextSummary = this.model.objectAt(currentSummaryIndex + 1);
 
-    if (this.recommendationSummaryIndex >= this.model.length) {
-      this.store.unloadAll('recommendation-summary');
-      yield timeout(Ember.testing ? 0 : 1000);
-      this.store.findAll('recommendation-summary');
+    if (nextSummary) {
+      this.transitionToSummary(nextSummary);
+    } else {
+      // This is a task because the accordion has actual timeouts for animation
+      yield timeout(0);
+      this.send('reachedEnd');
```

Comment:
This is probably just fine, but whenever I see a `timeout(0)` I wonder if a `run.next` or `run.scheduleAfter` would be more appropriate.

Reply:
yeah it’s weird… it doesn’t really need a timeout at all actually, I just got a linting error for not having a `yield`. What do you think of 32ecce79 instead to silence that linting error so there’s no need for a timeout? This is only a task because the accordion is a parallel `RecommendationCard` consumer that uses `timeouts` for the closing animation so `proceed` is the interface the card accepts 😐